civihr_default_theme/scss/components/_leave-calendar.scss

```diff
@@ -107,7 +107,26 @@ $chr-leave-calendar-legend-size:                  20px;
   font-size: $crm-font-size-small;
 }
 
-
 .chr_leave-calendar__weekend {
   opacity: 0.5;
 }
+
+.chr_leave-calendar__item--type--public-holiday {
+  background-color: #8ec68a;
+  border-color: #8ec68a;
+}
+
+.chr_leave-calendar__item--type--weekend {
+  background-color: #c7cbd0 !important;
+  border-color: #c7cbd0 !important;
+}
+
+.chr_leave-calendar__item--type--non-working {
+  background-color: #41afcb !important;
```

Comment:
Kept `!important` for `weekend` and `non-working`day to give them more priority. 
Example: If a date is weekend and also a leaveRequest is there for that date, showing colors for the `weekend` as leave is not calculated for that day.